### ISSUE TITLE
purpose of mask erosion

### ISSUE THREAD

**Issue by genesis02** [JustusThies/NeuralVoicePuppetry#23]
what is the purpose of mask erosion in forward part of the rendering network . 

    def maskErosion(self, mask, erosionFactor):
        offsetY = int(erosionFactor * 40)
        # throat
        mask2 = mask[:,:,0:-offsetY,:]
        mask2 = torch.cat([torch.ones_like(mask[:,:,0:offsetY,:]), mask2], 2)
        # forehead
        offsetY = int(erosionFactor * 8) #<<<<
        mask3 = mask[:,:,offsetY:,:]
        mask3 = torch.cat([mask3, torch.ones_like(mask[:,:,0:offsetY,:])], 2)
        mask = mask * mask2 * mask3

        offsetX = int(erosionFactor * 15)
        # left
        mask4 = mask[:,:,:,0:-offsetX]
        mask4 = torch.cat([torch.ones_like(mask[:,:,:,0:offsetX]), mask4], 3)
        # right
        mask5 = mask[:,:,:,offsetX:]
        mask5 = torch.cat([mask5,torch.ones_like(mask[:,:,:,0:offsetX])], 3)
        return mask * mask4 * mask5

What is the expected output and why  ? Concatenation part seems to be confusing? Shouldn't it be in  reverse order?  Please explain.

Also can provide a sample set of TARGET image and it corresponding uv map .

I have a sample similar to this

![00007](https://user-images.githubusercontent.com/46645972/102008056-cd3f7500-3d53-11eb-9b1f-b880a10059d8.jpg)
![7](https://user-images.githubusercontent.com/46645972/102008802-32499980-3d59-11eb-9bd7-5b95e82d8ff9.jpg)

If this is not proper sample set then please guide .  The uv map has been the uv-coordinate image generated from uv-coordinates.

Thanks

**Comment by JustusThies** [comment]
Hi,

this mask is used for the 'background' image, such that we remove e.g. moving chins or alike. The inpainting network has to fill this region with content that matches the 'inner' face region.

Best,
Justus Thies
